analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd

```diff
@@ -0,0 +1,591 @@
+---
+title: "TCGA PBTA comparison exploration"
+output: 
+  html_notebook:
+    toc: TRUE
+    toc_float: TRUE
+author: C. Savonen for ALSF CCDL
+date: 2020
+---
+
+#### Purpose: 
+Why are TCGA and PBTA data so different and why is TCGA TMB not higher than 
+PBTA as we expected from literature and the results from Grobner et al. 2019
+
+Related issues:
+[Original Issue 3](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/3)
+[Issue 257](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/257)
+[Draft PR 521](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/521)
+
+*Main three exploratory questions asked in this notebook:*
+
+1) [Is the tumor read depth different between PBTA and TCGA?](#how-does-the-tumor-sequencing-depth-compare-for-each-tcga-and-pbta-data)
+) 
+2) [Do TMB comparisons results change if we calculate TMB with each caller by itself?](#how-does-the-tmb-comparison-look-for-each-caller-by-itself)
+3) [How much do the TCGA and PBTA overlap in their target WXS regions?](#overlap-of-the-target-regions-of-both-datasets)
+
+#### Conclusions from this notebook: 
+Although it may be partially a caller-specific thing, we also suspect a problem with 
+Lancet's TCGA calls which is likely because it is all WXS data. 
+This issue is further investigated in the [Lancet WXS-WGS analysis notebook](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/lancet-wxs-tests/lancet-paired-WXS-WGS.Rmd) and the [Lancet padded-unpadded analysis notebook](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/lancet-wxs-tests/lancet-padded-vs-unpadded.Rmd). 
+
+#### Usage
+
+To run this from the command line, use:
+```
+Rscript -e "rmarkdown::render('analyses/snv-callers/lancet-wxs-tests/explore-tcga-pbta.Rmd', 
+                              clean = TRUE)"
+```
+
+_This assumes you are in the top directory of the repository._
+
+## Setup
+
+```{r}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+
+# We will need the calculate_tmb function from this script
+source(file.path("..", "..", "snv-callers", "util", "tmb_functions.R"))
+```
+
+Declare directory paths. 
+
+```{r}
+scratch_dir <- file.path("..", "..", "..", "scratch")
+data_dir <- file.path("..", "..", "..", "data")
+ref_dir <- file.path("..", "ref_files")
+plots_dir <- file.path("plots", "tcga-vs-pbta-plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir, recursive = TRUE)
+}
+```
+
+### Special functions 
+
+Function for setting up PBTA data from the database. 
+This database was originally created by the [run consensus bash script](https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/master/analyses/snv-callers/run_caller_consensus_analysis-pbta.sh)
+
+```{r}
+set_up_pbta <- function(data.name) {
+  # Given the name of dataset in the PBTA database, set up the TCGA data. 
```

Comment:
PBTA or TCGA here?  It seems like the function might be bets to be db agnostic, and just take a name and a database file; then it can collapse this and the later one, with just one argument added.

Reply:
I thought of doing this but the differences in metadata made it so I'd have to have weird/complicated arguments.

Reply:
Unless you have a suggestion for getting around that.

Reply:
It wasn't as weird as I thought it would be.